### PR TITLE
Add definition for Gusto office holidays

### PR DESCRIPTION
**Pull request by seemaullal**
In some cases, we only want to disable things (for example Chat) on holidays on which the Gusto office  is closed but not other holidays even if they are bank/postal holidays.

This came up because we disabled Chat yesterday (Columbus Day) even though the office was open and chat was open

### CODE REVIEW

**Comment by phanle** [comment]
I would like to know how you would plan to use in ZP.

File: lib/holidays/gusto.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          9 => [{:wday => 1, :week => 1, :name => "Labor Day", :regions => [:gusto]}],
+          11 => [{:wday => 4, :week => 4, :name => "Thanksgiving", :regions => [:gusto]},
+                {:wday => 5, :week => 4, :name => "Day After Thanksgiving", :regions => [:gusto]}],
+          12 => [{:mday => 25, :observed => lambda { |date| Holidays.to_weekday_if_weekend(date) }, :observed_id => "to_weekday_if_weekend", :name => "Christmas Day", :regions => [:us, :gusto]}],
+      }
+    end
+  end
+end
+
+Holidays.merge_defs(Holidays::Gusto.defined_regions, Holidays::Gusto.holidays_by_month)
```
Comment (phanle) [inline lib/holidays/gusto.rb:22]: Hey I am thinking that this specification is unique to gusto and should live in `zenpayoll` code base. Worth trying it?

**Comment by seemaullal** [comment]
@phanle This would be used in a place like https://github.com/Gusto/zenpayroll/blob/development/app/services/care/support_hour.rb like this:
`time.to_date.is_gusto_business_day?` and then here, https://github.com/Gusto/zenpayroll/blob/development/lib/date_extensions.rb we would have something like:

``` ruby
  def is_gusto_business_day?
    !weekend? && !holiday?(:gusto, :observed)
  end
```

We can try moving to `zenpayroll` as I guess it makes sense to be specific to Gusto. I put it in this gem because this is how we calculate holidays currently (and we have already added some customizations/monkey patching) and it has a nice implementation that we can use. 

We also have a pretty outdated version of this gem– the newer versions make it easy to create custom holiday logic with custom YAML files (the initial approach I took by adding a YAML file to the ZP app) but the version of the gem we are using does not support this. Will figure out another way to move the logic though and see how it works out.

**Comment by phanle** [comment]
@zp-alfred accept

**Comment by zp-alfred** [comment]
Accepted by phanle!

**Pull request closed without merging**